README.md

```diff
@@ -265,6 +265,14 @@ It's useful in several situations, e.g.:
 - Joining separate changes that make more sense when applied together
 - Rewriting _work in progress_ commits
 
+### When to avoid rebase or squash?
+
+Both rebase and squash commands substitutes existing commits for new ones. So it's not recommended to use with branches/commits that are being referenced by others or exists on the remote repository.
```

Comment:
I don't think neither that rebasing personal branches should be avoided.

My concern is just with rebase of shared branches and squash done by `git rebase -i`.

The Github's _Squash and Merge_ option when merging a Pull Request is ok, if the PR is not from a shared branch to another. Like `master <- develop`

The intention I wanted with my text, is just alert of the problem that can happen when we refer commits using `git rebase -i` inadvertently, because it's pretty easy to mistakenly refer commits that exists on shared branches.

That alone would at least change the SHA-1 of those commits and add conflicts between the branches whose fix could duplicate the commits in the best scenario.